frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { cephBlockPoolResource } from '../constants/resources';
+import { Dropdown } from '@patternfly/react-core/dist/js/components/Dropdown/Dropdown';
+import { DropdownItem } from '@patternfly/react-core/dist/js/components/Dropdown/DropdownItem';
+import { DropdownToggle } from '@patternfly/react-core/dist/js/components/Dropdown';
+import { CaretDownIcon } from '@patternfly/react-icons';
+
+const PoolResourceComponent: React.FC<PoolParams> = ({ onParamChange }) => {
+  const [data, loaded, loadError] = useK8sWatchResource<K8sResourceKind[]>(cephBlockPoolResource);
+  const [isOpen, setOpen] = React.useState<boolean>(false);
+  const [poolName, setPoolName] = React.useState('');
+
+  const handleDropdownChange = (e) => {
+    setPoolName(e.currentTarget.id);
+    onParamChange(e.currentTarget.id);
+  };
+
+  const poolDropdownItems = _.map(data, (pool, key) => (
+    <DropdownItem
+      key={key}
+      component="button"
+      id={pool?.metadata?.name}
+      onClick={(e) => handleDropdownChange(e)}
+    >
+      {pool?.metadata?.name}
+    </DropdownItem>
+  ));
+
+  const onToggle = () => {
+    setOpen(!isOpen);
+  };
+  const onSelect = React.useCallback(() => {
+    setOpen(false);
+  }, [setOpen]);
```

Comment:
Is it explicitly required? I thought dropdowns has this functionality.

Reply:
Yes,  dropdown handles it. But here I feel she has used this as its being used for `onToggle`. @a2batic Why do we need `DropDownToggle` here?

Reply:
Patternfly dropdown's don't

Reply:
yes, using patternfly dropdown, so these are not implicitly handled.